src/backlight/strategies/filter.py

```diff
@@ -49,11 +49,14 @@ def skip_entry_by_spread(
     assert trades.index.unique().isin(mkt.index).all()
 
     spread = mkt.spread
-
     deleted_ids = []  # type: List[int]
-    for i in trades.ids:
-        entry = trades.get_trade(i).index[0]
+
+    sorted_by_ids = (
+        trades.reset_index().sort_values(by=["_id", "index"]).set_index("index")
+    )
+    for i in range(0, sorted_by_ids.index.size, 2):
```

Comment:
Here, you assumed that each trade always contains entry and exit? It is not always true.